src/build/env.py

```diff
@@ -1,312 +1,293 @@
+        # Make the virtual environment's scripts available to the project's build dependecies.
+        path = environ.get('PATH')
+        environ['PATH'] = os.pathsep.join([self._scripts_dir, path]) if path is not None else self._scripts_dir
 
-    :param path: The path where to create the isolated build environment
-    :return: The Python executable and script folder
-    """
-    cmd = [str(path), '--no-setuptools', '--no-wheel', '--activators', '']
-    result = virtualenv.cli_run(cmd, setup_logging=False)
-    executable = str(result.creator.exe)
-    script_dir = str(result.creator.script_dir)
-    return executable, script_dir
-
-
-@functools.lru_cache(maxsize=None)
-def _fs_supports_symlink() -> bool:
-    """Return True if symlinks are supported"""
-    # Using definition used by venv.main()
-    if os.name != 'nt':
-        return True
+        return environ
 
-    # Windows may support symlinks (setting in Windows 10)
-    with tempfile.NamedTemporaryFile(prefix='build-symlink-') as tmp_file:
-        dest = f'{tmp_file}-b'
-        try:
-            os.symlink(tmp_file.name, dest)
-            os.unlink(dest)
-            return True
-        except (OSError, NotImplementedError, AttributeError):
-            return False
-
-
-def _create_isolated_env_venv(path: str) -> Tuple[str, str]:
-    """
-    On Python 3 we use the venv package from the standard library.
-
-    :param path: The path where to create the isolated build environment
-    :return: The Python executable and script folder
-    """
-    import venv
-
-    import packaging.version
+    @property
+    def python_executable(self) -> str:
+        return self._python_executable
 
-    venv.EnvBuilder(with_pip=True, symlinks=_fs_supports_symlink()).create(path)
-    executable, script_dir, purelib = _find_executable_and_scripts(path)
+    @staticmethod
+    def log(message: str) -> None:
+        if sys.version_info >= (3, 8):
+            _logger.log(logging.INFO, message, stacklevel=2)
+        else:
+            _logger.log(logging.INFO, message)
 
-    # Get the version of pip in the environment
-    pip_distribution = next(iter(metadata.distributions(name='pip', path=[purelib])))  # type: ignore[no-untyped-call]
-    current_pip_version = packaging.version.Version(pip_distribution.version)
 
-    if platform.system() == 'Darwin' and int(platform.mac_ver()[0].split('.')[0]) >= 11:
-        # macOS 11+ name scheme change requires 20.3. Intel macOS 11.0 can be told to report 10.16 for backwards
-        # compatibility; but that also fixes earlier versions of pip so this is only needed for 11+.
-        is_apple_silicon_python = platform.machine() != 'x86_64'
-        minimum_pip_version = '21.0.1' if is_apple_silicon_python else '20.3.0'
-    else:
-        # PEP-517 and manylinux1 was first implemented in 19.1
-        minimum_pip_version = '19.1.0'
+class IsolatedEnvManager(Generic[IsolatedEnvType]):
```

Comment:
Both are fairly generic; "builder" has a more precise meaning in `ProjectBuilder` and using the same term to mean two different things as part of the same API can be confusing.

Reply:
> And I am starting to think this would be better as a `classmethod` context manager in the implementation class.

I agree, but I was under the impression that we didn't want to expose the default isolated env because we consider using pip to be an 'implementation detail'.

Reply:
I think it was just so that to get people to use `IsolatedEnvBuilder` instead IIRC.

Reply:
I am looking at your changes and I think you misunderstand me. The context manager should not be the instance, but instead a `classmethod` constructor. See the example I linked above. This should allow us to move the `create` and `prepare_environ` contents to `__init__`.

Reply:
That's a bit cleaner I suppose though I generally don't like having side effects in `__init__`.

Reply:
Yeah, I'd prefer to keep `create` separate - I really don't want to go about performing I/O in the constructor.  The trouble is that you'd then need to guard the object's methods prior to the creation of the env.  If you think about it, a proxy object like `IsolatedEnvManager` is the obvious solution to this problem.